src/pages/index.vue

```diff
@@ -53,72 +57,67 @@ const props = defineProps({
         default: ""
     }
 });
+type PropName = keyof typeof props;
 
 const indicatorNames = computed(() => (appConfig.value ? Object.keys(appConfig.value.indicators) : {}));
 const unknownProps: Ref<string[]> = ref([]);
 
-const notFoundMsg = (valueType, value) => `Unknown ${valueType}: ${value}.`;
+const notFoundMsg = (valueType: string, value: string) => `Unknown ${valueType}: ${value}.`;
 
 const notFoundDetail = computed(() => {
     return unknownProps.value.map((propName) => notFoundMsg(propName, props[propName])).join(" ");
 });
 
-const countryToSelect: Ref<null | string> = ref(null);
+const checkRouteProp = (propName: PropName, candidates: string[]) => {
+    // Do case-insensitive check against route prop - add to unknown props if not found
+    if (props[propName] === "") return "";
+    const pattern = new RegExp(`^${props[propName]}$`, "i");
+    return candidates.find((i) => pattern.test(i));
+};
 
 const selectDataForRoute = async () => {
-    if (!appConfig.value) {
-        return;
-    }
+    if (!appConfig.value) return;
 
     const unknown = [];
-    const checkRouteProp = (propName: "pathogen" | "version" | "indicator" | "country", candidates: string[]) => {
-        if (!props[propName]) {
-            return "";
-        }
-
-        // Do case-insensitive check against route prop - add to unknown props if not found
-        const pattern = new RegExp(`^${props[propName]}$`, "i");
-        const result = candidates.find((i) => pattern.test(i));
-        if (!result) {
-            unknown.push(propName);
-        }
-        return result;
+    const possibleValuesForProps: Record<PropName, string[]> = {
+        pathogen: [PATHOGEN],
+        version: [VERSION],
+        indicator: Object.keys(appConfig.value.indicators),
+        country: appConfig.value.countries
     };
-
-    checkRouteProp("pathogen", [PATHOGEN]);
-    checkRouteProp("version", [VERSION]);
-    const indicator = checkRouteProp("indicator", Object.keys(appConfig.value.indicators));
-    const country = checkRouteProp("country", appConfig.value.countries);
+    Object.keys(possibleValuesForProps).forEach((prop: PropName) => {
+        if (checkRouteProp(prop, possibleValuesForProps[prop]) === undefined) {
+            unknown.push(prop);
+        }
+    });
 
     unknownProps.value = unknown;
+    if (unknownProps.value.length) return;
+
+    let path = "";
+    path += `/${props.pathogen || PATHOGEN}`;
+    path += `/${props.version || VERSION}`;
+    path += `/${props.indicator || indicatorNames.value[0]}`;
+    const currentPath = router.currentRoute.value.path;
+    if (currentPath.indexOf(path) !== 0) {
+        router.replace(path);
+    }
```

Comment:
since any number of these props can be empty strings, we construct the url with default values if missing and see if that is a prefix of the current path, if not then we replace the route.

Reply:
I find this refactor quite confusing. I don't really get the benefit of building the "base path OR requested path" WITH the assumption that either is fine because you've already checked for unknowns. I get that the nesting in the previous implementation was  gross, but at least it was reasonably clear what case you were dealing with in each arm - this just seems to attenuate the confusion.. And then having to check values a second time to sort out the casing doesn't seem optimal either.

I don't have any brilliant suggestions for what to replace it with. Maybe just some more comments would help. But it would definitely be good to avoid the double check if possible.

Reply:
i have found a way to avoid the double check by just storing the checked versions in an object, will have a think about this path stuff, the thing is i found the previous implementation much more confusing because it just happened to work, what i mean by that is it was hinging on whether or not the indicator was empty, but it would not work if we had another disease or version of data, this type of logic will work for different diseases and versions

Reply:
i have thought about this and its a tricky problem, for now i have reverted it to similar path logic to old route checking but have created a ticket for it since this is a problem we havent solved, would be nice to do it properly at some point in the future

Reply:
Fair enough - yeah, feels like it should be easy enough to find a nice of expressing this bit of pure logic, but not so!